contrib/src/main/scala/akka/stream/contrib/Valve.scala

```diff
@@ -98,9 +105,17 @@ final class Valve[A](mode: SwitchMode) extends GraphStageWithMaterializedValue[F
           promise.success(succeed)
       }
 
+      val getModeCallback = getAsyncCallback[Promise[SwitchMode]](_.success(mode))
+
       override def flip(flipToMode: SwitchMode): Future[Boolean] = {
         val promise = Promise[Boolean]()
-        callback.invoke((flipToMode, promise))
+        flipCallback.invoke((flipToMode, promise))
+        promise.future
+      }
+
+      override def getMode(): Future[SwitchMode] = {
+        val promise = Promise[SwitchMode]()
+        getModeCallback.invoke(promise)
```

Comment:
Note that this will never complete if the stage was already stopped, or stops before the async callback gets invoked. There is a new `invokeWithFeedback()` that recently got merged in Akka, but is not yet released that solves the problem.

Reply:
Thanks for the tip @johanandren, very useful. So what do you suggest ? We can wait on this one until `invokeWithFeedback` is released

Reply:
Maybe put a `// FIXME will never complete promise if stage is stopped, use invokeWithFeedback when Akka 2.5.7 is released` in here at least

Reply:
done